alchemist/alchemist-fxui/src/main/kotlin/it/unibo/alchemist/boundary/monitors/AbstractFXDisplay.kt

```
@@ -0,0 +1,284 @@
+    /**
+     * Group dedicated for painting the background.
+     */
+    val background = Group()
+
+    init {
+        firstTime = true
+        setStep(steps)
+        val repaintOnResize = ChangeListener<Number> { _, _, _ ->
+            repaint()
+        }
+        widthProperty().addListener(repaintOnResize)
+        heightProperty().addListener(repaintOnResize)
+        effectsCanvas.isMouseTransparent = true
+        effectsCanvas.widthProperty().bind(widthProperty())
+        effectsCanvas.heightProperty().bind(heightProperty())
+        children.addAll(background, effectsCanvas, interactions.canvases)
+    }
+
+    override fun getViewStatus(): FXOutputMonitor.ViewStatus {
+        return this.viewStatus
+    }
+
+    override fun setViewStatus(viewStatus: FXOutputMonitor.ViewStatus) {
+        this.viewStatus = viewStatus
+    }
+
+    override fun getStep(): Int {
+        return this.step
+    }
+
+    /**
+     * {@inheritDoc}.
+     *
+     * @throws IllegalArgumentException if the step is not bigger than 0
+     */
+    @Throws(IllegalArgumentException::class)
+    override fun setStep(step: Int) {
+        if (step <= 0) {
+            throw IllegalArgumentException("The parameter must be a positive integer")
+        }
+        this.step = step
+    }
+
+    override fun isRealTime(): Boolean {
+        return this.realTime
+    }
+
+    override fun setRealTime(realTime: Boolean) {
+        this.realTime = realTime
+    }
+
+    override fun repaint() {
+        mutex.acquireUninterruptibly()
+        try {
+            if (mayRender.get() && isVisible && !isDisabled) {
+                mayRender.set(false)
+                Platform.runLater {
+                    commandQueue.forEach { it() }
+                    mayRender.set(true)
+                }
+            }
+            interactions.repaint()
+        } catch (e: UninitializedPropertyAccessException) {
+            // wormhole hasn't been initialized
```

This is a silenced exception and must be fixed. When does it happen? Why?

---

As far as I could tell, this exception doesn't occur. Removed the try-catch.